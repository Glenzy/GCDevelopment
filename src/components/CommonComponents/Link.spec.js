import React from 'react';
import { shallow } from 'enzyme';
import LinkItem from './Link';
import { Link } from 'react-router-dom';


let wrapper;

const onClick = jest.fn();
const defaultProps = {
  navigationLink:"https://google.com.au",
  onClick,
  text:"Google",
};

describe('the links component', ()=>{
  it('should call click handler when clicked', ()=>{
    // given
    givenLinkWithProps(defaultProps);
      // when
    wrapper.simulate('click');
    // then
    expect(onClick.mock.calls.length).toBe(1);
  });
  it('renders displays Link correctly', () => {
    // given
    givenLinkWithProps(defaultProps);
    //when
    //.shallow(<Button />);
    //then
    //expect(wrapper.find(Link).props().to).toEqual("Google");
    expect(wrapper).toMatchSnapshot();
  });
  it('renders applies the navigationLink correctly', () => {
    // given
    givenLinkWithProps(defaultProps);
    //when
    //then
    expect(wrapper.find(Link).prop("to")).toEqual('https://google.com.au');
  });
});

const givenLinkWithProps = ({onClick, navigationLink, text, id, classNames, title}) => wrapper = shallow(
  <LinkItem
    title={title}
    navigationLink={navigationLink}
    onClick={onClick}
    text={text}
    id={id}
    classNames={classNames}
  />
);
