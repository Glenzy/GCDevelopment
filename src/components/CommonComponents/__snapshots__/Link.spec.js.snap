// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the links component renders displays Link correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LinkItem
    classNames="navLink"
    id=""
    navigationLink="https://google.com.au"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    text="Google"
    title="Link title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Google",
      "className": "navLink",
      "id": "",
      "onClick": [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "replace": false,
      "title": "Link title",
      "to": "https://google.com.au",
    },
    "ref": null,
    "rendered": "Google",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Google",
        "className": "navLink",
        "id": "",
        "onClick": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "replace": false,
        "title": "Link title",
        "to": "https://google.com.au",
      },
      "ref": null,
      "rendered": "Google",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
